roxctl/sensor/generate/generate_test.go

```diff
@@ -322,7 +322,7 @@ func (s *sensorGenerateTestSuite) TestSlimCollectorSelection() {
 			serverHasKernelSupport: true,
 			slimCollectorFlag:      nil,
 			warning:                nil,
-			expectSlimMode:         true,
+			expectSlimMode:         false,
 		},
 		"No flags and no kernel support in central: default to full collector": {
 			serverHasKernelSupport: false,
```

Comment:
This `KernelSupport` thing at the time of Secured Cluster manifests _generation_ is another chunk which _could be_ cleaned up.
IIRC, the availability of Kernel Support means either the Central is connected to the internet and can reach our place for driver downloads or that there's a support package uploaded to Central.
For core_bpf Collector none of this matters and so some _generation_ code could be simplified.

Still, the old versions of `roxctl` which expect some API contract should continue working with the new Central so if there are kernel support flags they should be returned as `true`, etc.

Also, for as long as ACS versions that use collector drivers aren't deprecated, the driver download and the support package mechanisms should exist even in the new Centrals.

Reply:
I think all currently supported versions do not require the kernel support check, or the driver downloads / support package mechanism, so I think we're in a good place to remove them. I'll do this in a follow up PR.